doc/user-manual/language/sort-system.lagda.rst

```diff
@@ -0,0 +1,193 @@
+  univSort Setω     = Setω                      (only if --omega-in-omega is enabled)
+
+.. _piSort:
+
+piSort
+------
+
+Similarly, ``piSort s1 s2`` is a constructor that computes the sort of
+a Π-type given the sort ``s1`` of its domain and the sort ``s2`` of its
+codomain as arguments.
+
+To understand how ``piSort`` works in general, we set the following scenario:
+
+* ``sA`` and ``sB`` are two (possibly different) sorts.
+* ``A : sA``, meaning that ``A`` is a type that has sort ``sA``.
+* ``x : A``, meaning that ``x`` has type ``A``.
+* ``B : sB``, meaning that ``B`` is a type (possibly different than ``A``) that
+  has sort ``sB``.
+
+Under these conditions, we can build the dependent function type
+``(x : A) → B : piSort sA (λ x → sB)``. This type signature means that the
+dependent function type ``(x : A) → B`` has a (possibly unknown) but
+well-defined sort ``piSort sA sB``, specified in terms of the element
+``x : A`` and the sorts of its domain and codomain.
+
+We list below all the possible computations that ``piSort`` can perform:
+
+.. code-block::
+
+  piSort s1 (λ x → s2) = funSort s1 s2          (if x does not occur freely in s2)
+  piSort s1 (λ x → s2) = Setω                   (if x occurs freely in s2)
+
+All these constructors do not represent new sorts but instead, they compute
+to the right sort once their arguments are known.
+
+More examples:
+
+* ``piSort Level (λ l → Set l)`` evaluates to ``Setω``
+* ``piSort (Set l) (λ _ → Set l')`` evaluates to ``Set (l ⊔ l')``
+* ``univSort (Set l)`` evaluates to ``Set (lsuc l)``
+* The function type ``∀ {A} → ∀ {B} → B → A → B`` with normal form
+  ``{A : _9} {B : _7} → B → A → B`` evaluates to
+  ``piSort (univSort _9) (λ A → funSort (univSort _7)
+  (funSort _7 (funSort _9 _7)))``
+
+Note that ``funSort`` and ``piSort`` are total functions on sort. But
+``univSort`` is not always well-defined. Eg. without adding the
+``--omega-in-omega`` option, ``Setω`` does not have a ``univSort``
+(successor sort) since there is currently no next sort to ``Setω``.
+Any uses of ``univSort`` will lead to a 'has bigger sort' constraint that
+ensures the argument is not ``Setω``.
+
+.. _set-omega-plus-n:
+
+Sorts ``Setωᵢ``
+---------------
+
+Agda implements sorts of the form ``Setωᵢ``, where ``i`` is a natural number
+``i : Nat``. These sorts constitute a second hierarchy ``Setωᵢ : Setωᵢ₊₁``
+where each sort has the type of its successor. This mechanism is similar to
+the one implemented in the standard hierarchy ``Setᵢ : Setᵢ₊₁`` that we
+introduced in the section :ref:`Universe Levels <universe-levels>`.
+
+But, unlike the standard hierarchy of universes ``Setᵢ``, this second
+hierarchy ``Setωᵢ`` does not support universe polymorphism. This means that
+it is not possible to quantify over *all* Setω+n at once. For example, the
+expression ``∀ {i} (A : Setω i) → A → A`` would not be a well-formed agda
+term.
+
+You can also refer to these sorts with the alternative syntax ``Setωi``.
```

Comment:
Is `Setω0` also legal (and the one where `0` is a subscript)?

Reply:
I would say so, but let's ask @jespercockx to double-check: are `Setω0` and `Setω₀` legal keywords after #4609 was merged? The behave is like their counterparts `Set0` and `Set₀` , right?

Reply:
They are legal but they are not keywords, you can rename `Setω` to anything such as `BleeBlorkl` in which case you can write `BleeBlorkl0` or `BleeBlorkl₀` and they mean the same thing as `BleeBlorkl` (which is to say, `Setω`).

Reply:
OK, then if `Set`, `Setω` and all its derivations are not keywords, I have to remove them from the list of keywords in [lexical structure](https://agda.readthedocs.io/en/latest/language/lexical-structure.html), right?

Reply:
Ah yes, I forgot to update that list in my PR. `Set` and `Prop` should indeed be removed from that list, though you don't necessarily have to add it to this PR (as @andreasabel said, it is better to keep PR's focused on one thing).

Reply:
OK perfect, i reverted the inclusion of `Setω` as keyword, you can do it for `Set` in yours.